src/bt-core-io.c

```diff
@@ -480,15 +480,14 @@ struct alarm_state {
 static enum ioresult
 alarm_event_in(int fd, void* data)
 {
-  uint64_t num_expirations;
-  ssize_t res;
-
   struct alarm_state* alarm_state = data;
   assert(alarm_state);
   if (alarm_state->cb)
     alarm_state->cb(alarm_state->data);
 
-  remove_timer(fd);
+  remove_fd_from_epoll_loop(fd);
+  if (TEMP_FAILURE_RETRY(close(fd)) < 0)
+    ALOGW_ERRNO("close");
   free(data);
```

Comment:
Just a very small nit here: this memory was allocated as 'alarm_state'. While you're add it, could you change the code to free alarm_state? It's not a real difference, but makes the connection a bit more obvious; I'd say.